src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,274 @@
+package duke.parser;
+
+import duke.commands.Command;
+import duke.commands.DoneCommand;
+import duke.commands.DeleteCommand;
+import duke.commands.ByeCommand;
+import duke.commands.AddCommand;
+import duke.commands.FindCommand;
+import duke.commands.ListCommand;
+import duke.commands.TasksOnCommand;
+
+import duke.exceptions.InvalidKeywordException;
+import duke.exceptions.NonExistentTaskException;
+import duke.exceptions.EmptyDescriptionException;
+import duke.exceptions.IncorrectFormatException;
+import duke.exceptions.InvalidNumberInputException;
+
+import duke.tasks.Deadline;
+import duke.tasks.Event;
+import duke.tasks.Task;
+import duke.tasks.ToDo;
+import duke.tasklist.TaskList;
+import duke.utils.Constants;
+import java.time.LocalDate;
+
+/**
+ * The Parser class encapsulates the parsing, interpretation and validation of the user's input commands
+ *
+ */
+public class Parser {
+
+    /**
+     * The parse method interprets the keyword of the user's input and accordingly calls the other task-specific
+     * parse functions
+     *
+     * @param input user's string input
+     * @param taskList the list of tasks to be operated upon
+     * @return a command returned by the respective task-specific parse function
+     */
+    public static Command parse(String input, TaskList taskList) {
+        String[] parsedInput = input.trim().split("\\s", 2);
+        String command = parsedInput[0];
+
+            switch(command.toUpperCase()) {
+            case "BYE":
+                return parseBye();
+            case "LIST":
+                return parseList(taskList);
+            case "DONE":
+                return parseDone(input, taskList);
+            case "DELETE":
+                return parseDelete(input, taskList);
+            case "TASKS_ON":
+                return parseTasksOn(input, taskList);
+            case "FIND":
+                return parseFind(input, taskList);
+            case "TODO":
+            case "DEADLINE":
+            case "EVENT":
+                return parseTask(input, taskList);
+            default:
+                throw new InvalidKeywordException();
+
+            }
+    }
+
+    /**
+     * The parseFind method is a task-specific parse function which is called when the user uses the
+     * 'find' keyword
+     *
+     * @param input user's string input
+     * @param taskList the list of tasks to be operated upon
+     * @return a FindCommand which will execute the task corresponding to the 'find' keyword
+     */
+    private static FindCommand parseFind(String input, TaskList taskList) {
+        try {
+            String[] parsedString = input.split("\\s", 2);
+            String word = parsedString[1].trim();
+            return new FindCommand(word);
+        } catch (ArrayIndexOutOfBoundsException ex) {
```

Comment:
I think it is best to avoid using exceptions for control flow.
How about a guard clause like
```java
if (parsedString.length < 2) {
    throw new EmptyDescriptionException("...");
}
```